doozer

```diff
@@ -829,6 +889,50 @@ def images_scan_for_cves(runtime):
     subprocess.check_call(["atomic", "scan"] + image_urls)
 
 
+@cli.command("images:show-tree", short_help="Display the image relationship tree")
+@click.option(
+    "--imagename", default=False, is_flag=True,
+    help="Use the image name instead of the dist-git name")
+@click.option(
+    "--yml", default=False, is_flag=True,
+    help="Ouput to yaml formatted text, otherwise generate a tree view")
+@pass_runtime
+def images_print(runtime, imagename, yml):
+    runtime.initialize(clone_distgits=False)
+
+    images = list(runtime.image_metas())
+
+    print_colors = ['green', 'cyan', 'blue', 'yellow', 'magenta']
+
+    def name(image):
+        return image.image_name if imagename else image.distgit_key
+
+    if yml:
+        color_print(yaml.safe_dump(runtime.image_tree, indent=2, default_flow_style=False), 'cyan')
+    else:
+        def print_branch(image, indent=0):
```

Comment:
I assume you have a Really Good Reason™ for using two nested functions in this action?

Reply:
Because I do that frequently and inner functions is a common python thing (in my experience). There is no use for them outside of the scope of show-tree so no reason to put them anywhere but inside show-tree. Plus it keeps the code DRY.
https://realpython.com/inner-functions-what-are-they-good-for/